clang/lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -1454,6 +1567,12 @@ static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
                                      QualType Owner2Type) {
   const auto *Owner2 = cast<Decl>(Field2->getDeclContext());
 
+  // In C23 mode, check for structural equivalence of attributes on the fields.
+  // FIXME: Should this happen in C++ as well?
```

Comment:
Can we get there in C++ at all? I wonder if you could assert the language mode is C
In C++ equivalence of types is limited https://eel.is/c++draft/temp.type